apolloschurchapp/src/tabs/home/Features/index.js

```diff
@@ -3,128 +3,149 @@ import { Query } from 'react-apollo';
-                sources: {
-                  uri: '',
-                },
-              },
-            },
-            {
-              id: 'fakeId4',
-              title: '',
-              subtitle: '',
-              parentChannel: {
-                id: '',
-                name: '',
-              },
-              coverImage: {
-                sources: {
-                  uri: '',
-                },
-              },
-            },
-          ]}
-        />
-      ) : (
-        get(features, 'userFeedFeatures', []).map(
-          ({ title, subtitle, actions, id }) =>
-            actions.length ? (
-              <ActionListCard
-                isLoading={loading}
-                key={id}
-                header={
-                  <>
-                    <StyledH6 numberOfLines={1}>{title}</StyledH6>
-                    <H3 numberOfLines={3}>{subtitle}</H3>
-                  </>
-                }
-                actions={actions}
-                onPressActionItem={({ action, relatedNode }) => {
-                  if (action === 'READ_CONTENT') {
-                    navigation.navigate('ContentSingle', {
-                      itemId: relatedNode.id,
-                      transitionKey: 2,
-                    });
+      get(features, 'userFeedFeatures', []).map(
+        ({ actions, cards, id, title, __typename, ...props }) => {
+          switch (__typename) {
+            case 'ActionListFeature':
+              return (
+                <ActionListCardFeature
+                  // TODO: How can we better handle generating a loading state.
+                  actions={loading ? actionListLoadingStateData : actions}
+                  isLoading={loading}
+                  onPressActionItem={({ action, relatedNode }) =>
+                    handleOnPressActionItem({
+                      action,
+                      navigation,
+                      relatedNode,
+                    })
                   }
-                  if (action === 'READ_EVENT') {
-                    navigation.navigate('Event', {
-                      eventId: relatedNode.id,
-                      transitionKey: 2,
-                    });
+                  onPressActionListButton={() =>
+                    handleOnPressCardActionButton({
+                      id,
+                      navigation,
+                      title,
+                    })
                   }
-                }}
-                onPressCardActionButton={() =>
-                  navigation.navigate('ContentFeed', {
-                    itemId: id,
-                    itemTitle: title,
-                  })
-                }
-              />
-            ) : null
-        )
+                  title={title}
+                  {...props}
+                />
+              );
+            case 'VerticalCardListFeature':
+              return (
+                <VerticalCardListFeature
+                  cards={cards.map((card) => ({
```

Comment:
Loading state idea.... what if this `Home/Features` component had a loading state, rather than deciding on a loading state from each of the component parts? 

Not for this PR long term - but might give us what we need. We'll never come up with a good one size fits all solution with a dynamic feed unless we ask the server for more info about what we'll eventually render.

Reply:
I mean it would still be an assumption on what to render just a in a different place. In some respects we can just fill the screen with placeholders for whatever. It would be something to at least look at while you waited. However, the idea behind a placeholder is that it represents the UI you're actually going to interact with. Which is to say that, ideally 🙃, after it has loaded things haven't moved around but instead have just become interactive. This is idealistic and not reality of course and it doesn't speak at all to a progressive loading/caching strategy. I just think it's important to know what we are aiming for. 

That said, I implemented it this way because that's how we typically implement loading states. I was also expecting that when you got around to progressively loading features this would be the best way to handle loadings states under those circumstances.